src/ansys_sphinx_theme/navbar_dropdown.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (C) 2021 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Navigation Dropdown for navigation bar."""
+
+import copy
+from functools import lru_cache
+import pathlib
+
+import bs4
+from docutils import nodes
+import sphinx
+from sphinx.util import logging
+from sphinx.util.nodes import make_refnode
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+def load_navbar_configuration(app: sphinx.application.Sphinx) -> None:
+    """Load the navbar configuration from a YAML file for the Sphinx app."""
+    if not (
+        "navigation_dropdown" in app.config.html_theme_options
+        and "layout_file" in app.config.html_theme_options["navigation_dropdown"]
+    ):
+        return
+
+    layout_file = app.config.html_theme_options["navigation_dropdown"].get("layout_file", None)
```

Comment:
```suggestion
    layout_file = app.config.html_theme_options["navigation_dropdown"]["layout_file"]
```
Since presence of `layout_file` is guaranteed at this point in the code.